releases/release-5.1.2.md

```diff
@@ -0,0 +1,152 @@
+    - Fix TiKV delete files before it ingests to RocksDB. [#10741](https://github.com/tikv/tikv/pull/10741)
+    - Fix the resolve failures caused by the left pessimisic locks. [#10653](https://github.com/tikv/tikv/pull/10653)
+
++ TiFlash
+
+    - Fix the issue of unexpected results when TiFlash fails to establish MPP connections
+    - Fix the potential issue of data inconsistency that occurs when TiFlash is deployed on multiple disks
+    - Fix a bug that MPP queries get wrong results when TiFlash server is under high load
+    - Fix a potential bug that MPP queries hang forever
+    - Fix the issue of concurrency problem between store initialization and `DDL`
+    - Fix a bug of incorrect results that occurs when queries contain filters like `CONSTANT` `<` | `<=` | `>` | `>=` `COLUMN`
+    - Fix the potential panic when applying `Snapshot` with multiple DDL operations
+    - Fix the issue that the store size in metrics is inaccurate under heavy writing
+    - Fix the potential issue that TiFlash cannot GC the delta data after running for a long time
+    - Fix the issue of wrong results when new collation is enabled
+    - Fix the potential panic issue that occurs when resolving locks
+    - Fix a bug that metrics display wrong value
+
++ PD
+
+    - Fix the bug that PD would not fix down-peer in time. [#4083](https://github.com/tikv/pd/pull/4083)
+    - Fix an issue where data is not stored when using max-replicas or location-labels to indirectly update default placement rule [#3915](https://github.com/tikv/pd/pull/3915)
+    - Fix the bug that PD may panic during scaling out TiKV. [#3911](https://github.com/tikv/pd/pull/3911)
+    - Fix the bug that hot region scheduler can not work when the cluster has evict leader scheduler. [#3697](https://github.com/tikv/pd/pull/3697)
+
++ Tools
+
+    + Backup & Restore (BR)
+
+        - fix the bug that the average speed isn't accurate in backup and restore [#1412](https://github.com/pingcap/br/pull/1412)
+
+    + Dumpling
+
+        - fix pending on show table status in some mysql version [#333](https://github.com/pingcap/dumpling/pull/333)
+
+    + TiCDC
+
+        - Fix json encoding may cause panic when processing a string type value in some cases. [#2783](https://github.com/pingcap/ticdc/pull/2783)
+        - Fix OOM when TiCDC captures too many regions [#2725](https://github.com/pingcap/ticdc/pull/2725)
+        - Fix gRPC keepalive error when memory pressure is high. [#2720](https://github.com/pingcap/ticdc/pull/2720)
+        - Fix a bug that causes TiCDC to panic on an unsigned tinyint [#2656](https://github.com/pingcap/ticdc/pull/2656)
+        - Fix open protocol, don't output an empty value when there is no change in one transaction. [#2621](https://github.com/pingcap/ticdc/pull/2621)
+        - Fixed a bug in DDL handling when the owner restarts. [#2607](https://github.com/pingcap/ticdc/pull/2607)
```

Comment:
```suggestion
        - Fix a bug that json encoding may cause panic when processing a string type value that is `string` or `[]byte`. [#2783](https://github.com/pingcap/ticdc/pull/2783)
        - Reduce gRPC window size to avoid OOM [#2725](https://github.com/pingcap/ticdc/pull/2725)
        - Fix gRPC keepalive error under high memory pressure. [#2720](https://github.com/pingcap/ticdc/pull/2720)
        - Fix a bug that an unsigned tinyint causes TiCDC to panic. [#2656](https://github.com/pingcap/ticdc/pull/2656)
        - Fix empty value issue in open protocol. An empty value is no longer output when there is no change in one transaction. [#2621](https://github.com/pingcap/ticdc/pull/2621)
        - Fix a bug in DDL handling during manual restarts. [#2607](https://github.com/pingcap/ticdc/pull/2607)
```